### PR TITLE
Custom erlang/dialyzer options

### PR DESCRIPTION
**Pull request by pinicarus**
Add an option to specify command line options for the Erlang dialyzer linter.

### CODE REVIEW

**Comment by pinicarus** [comment]
Hi,

Do I need to change something else or do you think this PR is OK for merging ?

Thanks a lot.

**Comment by pinicarus** [comment]
Hi @w0rp, any chance to get this PR merged ?

**Comment by tipishev-kivra** [comment]
A very welcome feature that I [forked](https://github.com/tipishev-kivra/ale). What needs to be done to get it merged?

**Comment by pinicarus** [comment]
Hi @hsanson,

Can you please check this PR ?

Thanks a lot.

File: doc/ale-erlang.txt [hunk 1]
```diff
@@ -13,6 +13,14 @@ g:ale_erlang_dialyzer_executable             *g:ale_erlang_dialyzer_executable*
   This variable can be changed to specify the dialyzer executable.
 
 
+g:ale_erlang_dialyzer_options                   *g:ale_erlang_dialyzer_options*
+                                                *b:ale_erlang_dialyzer_options*
+  Type: |String|
+  Default: `'-Wunmatched_returns -Werror_handling -Wrace_conditions -Wunderspec %s'`
```
Comment (hsanson) [inline doc/ale-erlang.txt:19]: Any particular reason to have %s as part of the options instead of harcoded in the command function like all other linters?

Comment (pinicarus) [reply doc/ale-erlang.txt]: It doesn't have to be, indeed.

I removed it

**Review by hsanson — DISMISSED** [review]
Good to merge after #3548 gets solved.

**Comment by hsanson** [comment]
ALE migrated from Travis to Github Actions (see #3548) so it is necessary to rebase this PR from latest master branch to trigger the required checks.

```
# Add dense-analysis as remote upstream on your local fork. Not 
# needed if already added.
git remote add upstream https://github.com/dense-analysis/ale.git

# Sync your local master with upstream master
git checkout master
git fetch upstream master
git merge upstream/master

# Rebase the PR branch with master
git checkout my-branch-name
git rebase master   # Fix any conflicts you may have

# Force push the updated PR branch to origin to trigger the checks
git push -f origin my-branch-name
```

**Comment by pinicarus** [comment]
@hsanson : branch has been rebased, all tests are green ;)

**Review by hsanson — APPROVED** [review]
Thanks for the patience. Good to merge.

**Pull request merged**